src/libs/histogram.c

```diff
@@ -333,29 +339,36 @@ static gboolean _lib_histogram_motion_notify_callback(GtkWidget *widget, GdkEven
 
   GtkAllocation allocation;
   gtk_widget_get_allocation(widget, &allocation);
-  if(d->dragging && d->highlight == 2)
+  if(d->dragging)
   {
-    float exposure = d->exposure + (event->x - d->button_down_x) * 4.0f / (float)allocation.width;
-    dt_dev_exposure_set_exposure(darktable.develop, exposure);
-  }
-  else if(d->dragging && d->highlight == 1)
-  {
-    float black = d->black - (event->x - d->button_down_x) * .1f / (float)allocation.width;
-    dt_dev_exposure_set_black(darktable.develop, black);
+    const float diff = dev->histogram_type == DT_DEV_HISTOGRAM_WAVEFORM ? d->button_down_y - event->y
+                                                                        : event->x - d->button_down_x;
+    const int range = dev->histogram_type == DT_DEV_HISTOGRAM_WAVEFORM ? allocation.height
+                                                                       : allocation.width;
+    if (d->highlight == 2)
+    {
+      float exposure = d->exposure + diff * 4.0f / (float)range;
```

Comment:
const

Reply:
Fixed in #4140 (where UI changes now are).